TestDeltaLakeFileOperations.testSelfJoin flaky

```
 Error:  io.trino.plugin.deltalake.TestDeltaLakeFileOperations.testSelfJoin -- Time elapsed: 0.556 s <<< FAILURE!
java.lang.AssertionError: 
Expected: 
		1 more occurrences of FileOperation[fileType=DATA, fileId=no partition, operationType=INPUT_FILE_NEW_STREAM]
	at org.testng.Assert.fail(Assert.java:94)
	at io.trino.testing.MultisetAssertions.assertMultisetsEqual(MultisetAssertions.java:52)
	at io.trino.plugin.deltalake.TestDeltaLakeFileOperations.assertFileSystemAccesses(TestDeltaLakeFileOperations.java:639)
	at io.trino.plugin.deltalake.TestDeltaLakeFileOperations.assertFileSystemAccesses(TestDeltaLakeFileOperations.java:630)
	at io.trino.plugin.deltalake.TestDeltaLakeFileOperations.testSelfJoin(TestDeltaLakeFileOperations.java:281)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

https://github.com/trinodb/trino/actions/runs/6344560785/job/17235464113?pr=19196#step:6:24700

--- a/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java
+++ b/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java
@@ -277,8 +277,13 @@ public class TestDeltaLakeFileOperations
     public void testSelfJoin()
     {
         assertUpdate("CREATE TABLE test_self_join_table AS SELECT 2 as age, 0 parent, 3 AS id", 1);
+        Session sessionWithoutDynamicFiltering = Session.builder(getSession())
+                // Disable dynamic filtering so that the second table data scan is not being pruned
+                .setSystemProperty("enable_dynamic_filtering", "false")
+                .build();
 
         assertFileSystemAccesses(
+                sessionWithoutDynamicFiltering,
                 "SELECT child.age, parent.age FROM test_self_join_table child JOIN test_self_join_table parent ON child.parent = parent.id",
                 ImmutableMultiset.<FileOperation>builder()
                         .add(new FileOperation(LAST_CHECKPOINT, "_last_checkpoint", INPUT_FILE_NEW_STREAM))

